app/models/input_template_renderer.rb

```diff
@@ -4,7 +4,7 @@ class UndefinedInput < ::Foreman::Exception
 
   class RenderError < ::Foreman::Exception
   end
-
+  include Rails.application.routes.url_helpers
```

Comment:
You're right, this was added as part of the PXE-less reprovision try, where I needed to redner the `foreman_url` inside the template,removed